packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1907,4 +1908,85 @@ describe("BnsConnection", () => {
 
     connection.disconnect();
   });
+
+  describe("getFeeQuote", () => {
+    it("works for send transaction", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+
+      const sendTransaction: SendTransaction = {
+        kind: "bcp/send",
+        creator: {
+          chainId: connection.chainId(),
+          pubkey: {
+            algo: Algorithm.Ed25519,
+            data: fromHex("aabbccdd") as PublicKeyBytes,
+          },
+        },
+        recipient: await randomBnsAddress(),
+        memo: `We ❤️ developers – iov.one`,
+        amount: defaultAmount,
+      };
+      const result = await connection.getFeeQuote(sendTransaction);
+      expect(result.tokens!.quantity).toEqual("0"); // ignore token type since it is free anyway
+      expect(result.gasPrice).toBeUndefined();
+      expect(result.gasLimit).toBeUndefined();
+
+      connection.disconnect();
+    });
+
+    it("works for other BNS transaction", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+
+      const registerBlockchainTransaction: RegisterBlockchainTx = {
+        kind: "bns/register_blockchain",
+        creator: {
+          chainId: connection.chainId(),
+          pubkey: {
+            algo: Algorithm.Ed25519,
+            data: fromHex("aabbccdd") as PublicKeyBytes,
+          },
+        },
+        chain: {
+          chainId: "wonderland" as ChainId,
+          production: false,
+          enabled: true,
+          name: "Wonderland",
+          networkId: "7rg047g4h",
+          mainTickerId: "WONDER" as TokenTicker,
+        },
+        codecName: "rules_of_wonderland",
+        codecConfig: `{ rules: ["make peace not war"] }`,
+      };
+      const result = await connection.getFeeQuote(registerBlockchainTransaction);
+      expect(result.tokens!.quantity).toEqual("0"); // ignore token type since it is free anyway
+      expect(result.gasPrice).toBeUndefined();
+      expect(result.gasLimit).toBeUndefined();
+
+      connection.disconnect();
+    });
+
+    it("throws for unsupported transaction kind", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+
+      const otherTransaction: UnsignedTransaction = {
+        kind: "other/kind",
+        creator: {
+          chainId: connection.chainId(),
+          pubkey: {
+            algo: Algorithm.Ed25519,
+            data: fromHex("aabbccdd") as PublicKeyBytes,
+          },
+        },
+      };
+      await connection
+        .getFeeQuote(otherTransaction)
+        .then(() => fail("must not resolve"))
+        .catch(error => expect(error).toMatch(/transaction of unsupported kind/i));
```

Comment:
Is there something similar to `chai-as-promised` we can use?

Reply:
The only thing I am aware of is this issue: https://github.com/jasmine/jasmine/issues/1625

Looks like strict error message equality can be archived but not the looser checks that allow slight variation of the implementation.